tests/js/client/aql/vector/aql-vector-agency-wrong-definition-cluster.js

```diff
@@ -0,0 +1,189 @@
+
+////////////////////////////////////////////////////////////////////////////////
+/// @brief test suite
+////////////////////////////////////////////////////////////////////////////////
+
+function VectorIndexCorrectDefinitionInAgencyTest() {
+    let collection;
+    const dimension = 500;
+    const seed = 12132390894;
+
+    return {
+        setUp: function() {
+            db._createDatabase(dbName);
+            db._useDatabase(dbName);
+
+            collection = db._create(collName, {
+                numberOfShards: 3
+            });
+
+            let docs = [];
+            let gen = randomNumberGeneratorFloat(seed);
+            for (let i = 0; i < 100; ++i) {
+                const vector = Array.from({
+                    length: dimension
+                }, () => gen());
+                docs.push({
+                    vector
+                });
+            }
+            collection.insert(docs);
+        },
+
+        tearDown: function() {
+            db._useDatabase("_system");
+            db._dropDatabase(dbName);
+        },
+
+        testVectorIndexCreationCorrect: function() {
+            let beforeVersion = IM.agencyMgr.get("Current/Version").arango.Current.Version;
+            collection.ensureIndex({
+                "name": "test",
+                "params": {
+                    "dimension": dimension,
+                    "metric": "l2",
+                    "nLists": 1,
+                },
+                "fields": ["vector"],
+                "type": "vector"
+            });
+
+            // wait for servers to pick up the changes
+            internal.sleep(5);
+
+            let afterVersion = IM.agencyMgr.get("Current/Version").arango.Current.Version;
+
+            assertTrue(afterVersion - beforeVersion < 10);
+        },
+    };
+}
+
+// This test confirms that even though there might be an incomplete definition of
+// vector index it will still not cause the loop of incorrect comparison between
+// indexes.
+// The relevant code for this is in IndexFactory::equal
+function VectorIndexInvalidDefinitionInAgencyTest() {
+    let collection;
+    const dimension = 500;
+    const seed = 12132390894;
+
+    return {
+        setUp: function() {
+            db._createDatabase(dbName);
+            db._useDatabase(dbName);
+
+            collection = db._create(collName, {
+                numberOfShards: 3
+            });
+
+            let docs = [];
+            let gen = randomNumberGeneratorFloat(seed);
+            for (let i = 0; i < 100; ++i) {
+                const vector = Array.from({
+                    length: dimension
+                }, () => gen());
+                docs.push({
+                    vector
+                });
+            }
+            collection.insert(docs);
+        },
+
+        tearDown: function() {
+            db._useDatabase("_system");
+            db._dropDatabase(dbName);
+        },
+
+        testVectorIndexCreationWithMissingField: function() {
```

Comment:
Could you add a comment which field is missing (and where in the definition)?

Reply:
It is `nProbes`, which is now set by default but wasn't in old versions. I think it would be a good idea to add a comment about this.